src/components/views/ViewDataTable/ViewDataTableColumnMenu.tsx

```diff
@@ -0,0 +1,65 @@
+import { MenuItem } from '@material-ui/core';
+import { FormattedMessage as Msg } from 'react-intl';
+import { forwardRef, MouseEventHandler } from 'react';
+import {
+    GridColDef,
+    GridColumnMenuContainer,
+    GridColumnMenuProps,
+    GridFilterMenuItem,
+} from '@mui/x-data-grid-pro';
+
+export type ViewDataTableColumnMenuProps = {
+    onConfigure: (colId : string) => void;
+    onDelete: (colId : string) => void;
+    onRename: (colId : string) => void;
+    showConfigureButton: (field: GridColDef['field']) => boolean;
+};
+
+const ViewDataTableColumnMenu = forwardRef<
+    HTMLUListElement,
+    GridColumnMenuProps & ViewDataTableColumnMenuProps
+>(function ViewDataTableColumnMenuComponent(props, ref) {
+    const { hideMenu, currentColumn, onConfigure, onDelete, onRename, ...rest } = props;
+
+    const onClickRename : MouseEventHandler = (ev) => {
+        ev.stopPropagation();
+        onRename(currentColumn.field);
+    };
+
+    const onClickConfigure : MouseEventHandler = (ev) => {
+        ev.stopPropagation();
+        onConfigure(currentColumn.field);
+    };
+
+    const onClickDelete : MouseEventHandler = (ev) => {
+        ev.stopPropagation();
+        onDelete(currentColumn.field);
+    };
+
+    return (
+        <GridColumnMenuContainer
+            ref={ ref }
+            currentColumn={ currentColumn }
+            data-testid={ `grid-filter-column-menu-${currentColumn.field}` }
```

Comment:
Do these actually have to be unique? Including the column ID requires the test to know it (which is fine, because it's mocked, but does mean the same data is stored in multiple places). I thought about it when I read the test, but at the time imagined the test ID to be coming from inside MUI (for some reason, obviously unreasonable). Now that I see it's in our control, I'm thinking we could make it only locally unique (without the suffix) and use descendent selectors in the tests. What do you think?